CRM/Contribute/Form/AbstractEditPayment.php

```diff
@@ -669,14 +669,14 @@ protected function assignPaymentInfoBlock() {
     return $title;
   }
 
-  protected function assignContactEmailDetails() {
+  protected function assignContactEmailDetails(): void {
     if ($this->getContactID()) {
       [$displayName] = CRM_Contact_BAO_Contact_Location::getEmailDetails($this->getContactID());
       if (!$displayName) {
         $displayName = civicrm_api3('contact', 'getvalue', ['id' => $this->getContactID(), 'return' => 'display_name']);
       }
-      $this->assign('displayName', $displayName);
     }
+    $this->assign('displayName', $displayName) ?? NULL;
```

Comment:
```suggestion
    $this->assign('displayName', $displayName ?? NULL);
```

But noting while this does get rid of the notice, the header text is then missing the name: `Use this form to submit an event registration on behalf of . A LIVE transaction will be submitted using the selected payment processor.` It was missing it before too, but the actual problem is it's trying to display text that doesn't make sense in this situation.